docs/mdbook/src/android/geojson-guide.md

```diff
@@ -0,0 +1,349 @@
+# Introduction
+
+This guide will teach you how to use [`GeoJsonSource`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-geo-json-source/index.html") by deep diving into [`GeoJSON`]("https://geojson.org/") file format.
+You will start with fundamentals of how a map renders data internally and why you should prefer [JSON]("https://en.wikipedia.org/wiki/JSON") format in storing geospatial data
+
+# Goals
+After finishing  this documentation you should be able to:
+1. understand how [`Style`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.maps/-style/index.html?query=open%20class%20Style"), [`Layer`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.layers/-layer/index.html?query=abstract%20class%20Layer"), and [`Data source`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-source/index.html?query=abstract%20class%20Source") interact with each other
+2. describe JSON files
+3. explore building blocks of `GeoJSON` data
+4. use `GeoJSON` files in constructing [`GeoJsonSource`]("https://maplibre.org/maplibre-native/android/api/-map-libre%20-native%20-android/org.maplibre.android.style.sources/-source/index.html?query=abstract%20class%20Source")s
+5. update data at runtime
+
+### 1. Styles, Layers, and Data source
+- `Style ` defines the visual representation of the map such as colors and appearance.
+- `Layer` controls how data should be presented to the user
+- `Data source`  holds actual data and provides layers with it
+
+Styles consist of collections of layers and data source. Layers reference data source. Hence they require a unique source ID when you construct them.
+It would be meaningless if we don't have any data to show, so we need know how to supply data through a data source.
+Firstly, we need to understand how to store data and pass it into a data source; therefore, we will discuss JSON in the next session.
+
+### 2. JSON
```

Comment:
I think we can assume that everyone knows JSON...

We can probably link to an explanation of GeoJSON elsewhere and focus on concepts that are specific to MapLibre Android.